ui/raidboss/data/06-ew/raid/p5s.ts

```diff
@@ -1,13 +1,210 @@
+import NetRegexes from '../../../../../resources/netregexes';
+import Outputs from '../../../../../resources/outputs';
+import { Responses } from '../../../../../resources/responses';
 import ZoneId from '../../../../../resources/zone_id';
 import { RaidbossData } from '../../../../../types/data';
 import { TriggerSet } from '../../../../../types/trigger';
 
-export type Data = RaidbossData;
+const directions = {
+  'NE': true,
+  'SE': true,
+  'SW': true,
+  'NW': true,
+};
+
+export interface Data extends RaidbossData {
+  target?: string;
+  topazRays: { [time: number]: (keyof typeof directions | undefined)[] };
+}
+
+const convertAbilityIdToTopazRayIndex = (id: string): number => {
+  // 7703 is the Topaz Ray cast with the lowest cast time
+  return parseInt(id, 16) - parseInt('7703', 16);
+};
+
+const convertCoordinatesToDirection = (x: number, y: number): keyof typeof directions | undefined => {
+  if (x > 100)
+    return y < 100 ? 'NE' : 'SE';
+  if (x < 100)
+    return y < 100 ? 'NW' : 'SW';
+  return undefined;
+};
 
 const triggerSet: TriggerSet<Data> = {
   zoneId: ZoneId.AbyssosTheFifthCircleSavage,
   timelineFile: 'p5s.txt',
-  triggers: [],
+  initData: () => {
+    return {
+      topazRays: {},
+    };
+  },
+  triggers: [
+    {
+      id: 'P5S Attack',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '7A0E', source: 'Proto-Carbuncle' }),
+      run: (data, matches) => data.target = matches.target,
+    },
+    {
+      id: 'P5S Provoke',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '1D6D' }),
+      run: (data, matches) => data.target = matches.source,
+    },
+    {
+      id: 'P5S Sonic Howl',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '7720', source: 'Proto-Carbuncle', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P5S Venomous Mass',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '771D', source: 'Proto-Carbuncle', capture: false }),
+      alarmText: (data, _matches, output) => {
+        if (data.role === 'tank' && data.target !== data.me)
+          return output.tankSwap!();
+      },
+      alertText: (data, _matches, output) => {
+        if (data.role === 'tank' && data.target !== data.me)
+          return;
+
+        if (data.target === data.me)
+          return output.busterOnYou!();
+        return output.busterOnTarget!({ player: data.ShortName(data.target) });
+      },
+      outputStrings: {
+        busterOnYou: Outputs.tankBusterOnYou,
+        busterOnTarget: Outputs.tankBusterOnPlayer,
+        tankSwap: Outputs.tankSwap,
+      },
+    },
+    {
+      // Maybe too noisy?
```

Comment:
I was thinking just output.tankBusters for Venomous Gale would be fine for healer role.

Reply:
Sure! It has to be convoluted because they don't cast on a target. I'll add a note about that as well.